### PR TITLE
Create a multi-process test of the aerosol physics in mam_refactor.

### PR DESCRIPTION
**Pull request by overfelt**
Based on the amicphys source in mam_refactor.
This calls most of the processes we now have in mam4xx to pass the output from one to the in put of the next.

### CODE REVIEW

**Comment by jeff-cohere** [comment]
From the PNNL CI test log:

```
test 8
      Start  8: mam4_amicphys_1gridcell_tests
8: Test command: /usr/bin/sh "-c" "/people/svceagles/gitlab/4142/haero_Release_double/build/bin/test-launcher -- ./mam4_amicphys_1gridcell_tests --use-colour no"
8: Environment variables: 
8:  OMP_NUM_THREADS=1
8: Test timeout computed to be: 1500
8: Calling initialize_kokkos
8: Warning: command line argument '--kokkos-num-devices' is deprecated. Use '--kokkos-map-device-id-by=mpi_rank' instead. Raised by Kokkos::initialize().
8:  ExecSpace name: Cuda
8:  ExecSpace initialized: yes
8:  active avx set: 
8:  compiler id: GCC
8:  FPE support is enabled, current FPE mask: 0 (NONE)
8:  #host threads: 1
8: 
8: qaer4[0][1] != Approx(check_qaer4[0][1])): -6.5906690364983e-20 != 2.0066181180847e-12
8: 
8: ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
8: mam4_amicphys_1gridcell_tests is a Catch v2.13.8 host application.
8: Run with -? for options
8: 
8: -------------------------------------------------------------------------------
8: clear
8: -------------------------------------------------------------------------------
8: /people/svceagles/gitlab/4142/haero_Release_double/src/tests/mam4_amicphys_1gridcell.cpp:1755
8: ...............................................................................
8: 
8: /people/svceagles/gitlab/4142/haero_Release_double/src/tests/mam4_amicphys_1gridcell.cpp:2005: FAILED:
8:   REQUIRE( qaer4[i][j] == Approx(check_qaer4[i][j]).epsilon(epsilon) )
8: with expansion:
8:   -0.0 == Approx( 0.0 )
8: 
8: ===============================================================================
8: test cases:   4 |   3 passed | 1 failed
8: assertions: 282 | 281 passed | 1 failed
8: 
8: NO RESOURCE SPECS
8: RUN: OMP_PROC_BIND=spread OMP_PLACES=threads ./mam4_amicphys_1gridcell_tests --use-colour no
8: FROM: /qfs/people/svceagles/gitlab/4142/haero_Release_double/build/src/tests
 8/73 Test  #8: mam4_amicphys_1gridcell_tests .........................***Failed    0.26 sec
```

**Comment by overfelt** [comment]
Odd, I am testing with Cuda 11.2.2 / gcc 10.3.0 with no errors.   I'll look into it.

File: src/mam4xx/gasaerexch.hpp [hunk 1]
```diff
@@ -561,25 +572,35 @@ void mam_gasaerexch_1subarea(
...
+          qaer[n] = qaer_cur[iaer][n];
       }
       gasaerexch::mam_gasaerexch_1subarea_1gas_nonvolatile(
           dt, netprod, uptkaer_igas, qgas_cur[igas], qgas_avg[igas], qaer);
-      for (int n = 0; n < num_mode; ++n) {
-        qaer_cur[iaer][n] = qaer[n];
+
+      for (ModeIndex mode : GasAerExch::Modes()) {
+        const int n = static_cast<int>(mode);
+        if (1 || mode_contains_species(mode, aer))
```
Comment (jeff-cohere) [inline src/mam4xx/gasaerexch.hpp:596]: What is done by the `1 ||` part of this predicate?

File: src/mam4xx/nucleation.hpp [hunk 2]
```diff
@@ -503,7 +503,7 @@ void newnuc_cluster_growth(Real ratenuclt_bb, Real cnum_h2so4, Real cnum_nh3,
     tmpa = max(tmpa, 0.0);
 
     // tmpb = h2so4 gas diffusivity ([m2/s], then [m2/h])
-    tmpb = 6.7037e-6 * pow(temp_in, 0.75) / cair;
+    tmpb = 6.7037e-9 * pow(temp_in, 0.75) / cair;
```
Comment (jeff-cohere) [inline src/mam4xx/nucleation.hpp:506]: What is this numerical factor? If it's going to absorb an additional `1e-3` to accommodate the omission above, should we separate this into that conversion factor and give a name to the `6.7e-6` portion?

**Review by jeff-cohere — APPROVED** [review]
Looks good. Let's see if we can figure out what the PNNL systems don't like, if that's practical. Do you have an account on Deception, @overfelt ?

**Comment by overfelt** [comment]
@jeff-cohere , a PNNL login token is in the mail.  FedEx I'm told.

**Pull request merged**